webapp/packages/core-blocks/src/FormControls/useInputAutocomplete.ts

```diff
@@ -0,0 +1,187 @@
+    sourceHints,
+    searchFields: SEARCH_FIELDS,
+    matchStrategy,
+    predicate,
+  });
+  const menuRef = useRef<IMenuState>();
+  const state = useObservableRef(
+    () => ({
+      position: { x: 0, y: 0 } as IContextMenuPositionCoords,
+      inputValue: '',
+      isFound: false,
+      selectionStart: 0 as number | null,
+      replaceCurrentWord(replacement: string) {
+        const cursorPosition = this.selectionStart;
+        const words = this.inputValue.split(' ');
+
+        if (!this.currentWord || !isNotNullDefined(words) || !isNotNullDefined(cursorPosition)) {
+          return;
+        }
+
+        const start = cursorPosition - this.currentWord.length;
+        const end = cursorPosition;
+
+        this.inputValue = this.inputValue.slice(0, start) + replacement + this.inputValue.slice(end);
+        this.selectionStart = start + replacement.length;
+
+        if (this.inputRef.current) {
+          this.inputRef.current.value = this.inputValue;
+          this.inputRef.current.focus();
+          this.inputRef.current.setSelectionRange(this.selectionStart, this.selectionStart);
+        }
+
+        this.isFound = true;
+      },
+      get currentWord(): string {
+        const cursorPosition = this.selectionStart;
+
+        if (!cursorPosition) {
+          return '';
+        }
+
+        const substring = this.inputValue.slice(0, cursorPosition);
+
+        if (!substring) {
+          return '';
+        }
+
+        return substring.split(' ').at(-1) ?? '';
+      },
+      get proposals() {
+        if (this.isFound || !this.currentWord) {
+          return [];
+        }
+
+        return this.search.searchResult;
+      },
+    }),
+    {
+      proposals: computed,
+      selectionStart: observable.ref,
+      isFound: observable.ref,
+      currentWord: computed,
+      replaceCurrentWord: action.bound,
+      position: observable.ref,
+      inputValue: observable.ref,
+    },
+    { inputRef, search, menuRef },
+  );
+
+  const handleInput = useMemo(
+    () =>
+      debounce((event: Event) => {
+        const target = event.target as HTMLInputElement;
+
+        state.selectionStart = target.selectionStart;
+        state.inputValue = target.value;
+        state.isFound = false;
+        state.search.setSearch(state.currentWord);
+      }, INPUT_DELAY),
+    [state],
+  );
+
+  function handleKeyDown(event: any) {
+    switch (event.key) {
+      case 'Escape':
+        state.menuRef.current?.hide();
+        break;
+      case 'ArrowDown':
+      case 'ArrowUp':
+        state.menuRef.current?.first();
+        break;
+      default:
+        break;
+    }
+  }
+
+  useEffect(() => {
+    const input = state.inputRef.current!;
+
+    input.addEventListener('input', handleInput);
+    input.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      input.removeEventListener('keydown', handleKeyDown);
+      input.removeEventListener('input', handleInput);
+    };
+  }, []);
```

Comment:
This will lead to unexpected behavior if the input is unmounted because the hook is executed only once at the first render. I will suggest to remove deps ([]) to run this hook every re-render